src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1368,11 +1376,16 @@
                                    (update-in node-map [[condition env]] conj node-id)
                                    (assoc node-map [condition env] [node-id])))
                                {}
-                               condition-to-node-ids)]
+                               condition-to-node-ids)
+
+        ;; We sort the alpha nodes by the sum of the node ids they correspond to
+        ;; in order to make the order of alpha nodes for any given type consistent.
+        condition-to-node-entries (sort-by (fn [[k v]] (apply + v))
```

Comment:
I know I've already talked about this once, but again I have thought of something.

What if two node ids end up adding up to the same thing?  I think it'd be nondeterministic still then.
Like what if you had `[1 2 3]` or `[2 4]` or `[6]` etc. I think you get it.

Would a better `sort-by` be something that actually ensures they aren't the same?
```clj
(sort-by (fn [[_ v]] (sort v)) <etc>)
```

Obviously, this isn't quite as fast, but probably doesn't matter since there wouldn't be too many node ids per condition in practical cases.